api-http/proposals/snapshot-api.md

```diff
@@ -0,0 +1,566 @@
+# Snapshot Scheduling API Design specs
+
+:bulb: Following document embodies API proposal in relation to the following github issue https://github.com/AntelopeIO/leap/issues/396.
+
+## :beginner: Design Info
+
+### Preface
+It was requested that snapshot functionality and, in particular, snapshot API needs to be extended to support additional functionality. In particular, ability to schedule snapshot(s), edit schedule and query snapshot status.
+
+Following goals for an API were defined in mentioned above github issue:
+- Scenario 1: Create a snapshot for a specific block num
+- Scenario 2: Create a snapshot every X number of blocks
+
+Following considerations/design decisions were also expressed:
+- Ability to query existing requests and cancel them
+- Do pending requests persist across nodeos restarts?
+
+---
+
+
+:small_blue_diamond:Status:
+- [X] In progress
+- [ ] In review. Reviewer:
+- [ ] Finished
+---
+## :wrench: Proposal API additions
+
+
+| **API URL** 	| **Purpose** 	| **Parameters** 	|
+|----------	|----------	|----------	|
+| /producer/schedule_snapshot         	| Adds a task to perform snapshot         	| start_block, end_block, blocks_count, snapshot_description     	|
+| /producer/get_snapshot_requests         	| Returns scheduled snapshot requests         	| start_snapshot_request_id, limit (number of snapshots to return)         	|
+| /producer/get_snapshot_request_status         	| Queries status of a snapshot request        	| Snapshot ID         	|
+| /producer/unschedule_snapshot         	| Removes previously scheduled snapshot task         	| Snapshot ID         	|
+
+### Scheduling Snapshot
```

Comment:
Can number of requests be limited to avoid exhausting the disk space?

Reply:
The producer API is meant to be called by the operator of the node only, so there isn't really a security issue with no having a limit. That said, we could enforce some hard-coded high limit on the number of active requests.

Reply:
Great idea, thank you, added to document. Wont hurt for sure.